core/include/detray/field/constant_magnetic_field.hpp

```diff
@@ -15,12 +15,12 @@
 
 namespace detray {
 
-template <typename context_t = dindex>
+template <typename scalar_t, typename context_t = dindex>
 class constant_magnetic_field {
 
     public:
-    using vector3 = __plugin::vector3<scalar>;
-    using point3 = __plugin::point3<scalar>;
+    using vector3 = __plugin::vector3<scalar_t>;
```

Comment:
This might be what @krasznaa you mentioned during the chat? 
plugin-specific type should also be templated if I am correct

Reply:
Yes. In this class's case I would imagine making it templated on the `vector3` and `point3` types. Like:

```c++
template <typename vector3_t, typename point3_t, typename context_t = dindex>
class constant_magnetic_field {
```

Since the scalar type is only used for defining these two types in this class, we can even forego declaring those types as templates. So, we would specify a concrete type like:

```c++
using b_field_t = constant_magnetic_field<algebra::eigen::vector3<float>, algebra::eigen::point3<float> >;
```

I didn't try any of this out, but I **think** something like that could work.